src/controllers/controllerpreset.h

```diff
@@ -43,16 +45,38 @@ class ControllerPreset {
      * @param filename Name of the XML file to add
      * @param functionprefix Function prefix to add
      */
-    void addScriptFile(QString filename, QString functionprefix,
-                       bool builtin=false) {
+    void addScriptFile(QString filename, QString functionprefix, bool builtin = false) {
         ScriptFileInfo info;
         info.name = filename;
         info.functionPrefix = functionprefix;
+        // Always try to load script from the mapping's directory first
+        info.file = QFileInfo(dirPath().absoluteFilePath(filename));
         info.builtin = builtin;
-        scripts.append(info);
+        m_scripts.append(info);
         setDirty(true);
     }
 
+    QList<ScriptFileInfo> getScriptFiles(const QString& fallbackPath = QString()) const {
```

Comment:
I am doubting that this function should exist. Why not do all this in `addScriptFile`? Doing all this every time the ControllerPreset's scripts are used is superfluous.

Reply:
This will take more refactoring, but I think it will be worth it.

Reply:
Done.